### ISSUE TITLE
BootstrapHandler loads all webpack chunks even intended for dynamic loading.

### ISSUE THREAD

**Issue by xdenser** [vaadin/flow#10579]
How to reproduce:

just use 
`var dynamicImport = await import('dynamic-module');`
in the code.
And see that 'dynamic-module' is loaded immediatelly with boostrap page.
Did not test it with Vaadin15+ way of boostraping.
Is BootstrapHandler still supported in Vaadin19? It should be if it is public class in jar.

**Comment by Legioth** [comment]
How do I reproduce this? Where should I put the line of code with the dynamic import?

**Comment by xdenser** [comment]
in js/ts file in frontendfolder.

**Comment by xdenser** [comment]
And a hit how to fix this:  
see webpack documentation 
https://webpack.js.org/api/stats/
BootstrapHandler looks only in field "assetsByChunkName"
but there is also "initial":boolean field in "chunks" array.
If this is true then module should be loaded during boostrap, otherwise this is dynamic module and webpack should load it in runtime.

**Comment by pleku** [comment]
> Is BootstrapHandler still supported in Vaadin19? It should be if it is public class in jar.

Yes, but since Vaadin 15+ the modules are injected in a different way, writing the initial chunk import to the index.html page and that chunk loads all the other chunks - there are several since even the client engine is now processed by webpack.

When I add the `initial` flag to our stats plugin, only one chunk, with id-0, gets the `"initial": "true"` property in json. `HtmlWebpackPlugin` injects a `script`loading this chunk to the index.html. But I see that in addition to that, the chunk-0 loads the chunks 1-3 too even though those have `"initial": "false"`. But I'm quite sure that those would have to be loaded anyway after first chunk. And maybe this is doing what you are describing - loading non-initial dynamic chunks when needed? I have not found the place which decides what other chunks are loaded by the chunk-0, but at least it is not loading everything as there are actually more chunks.

For Flow 2.x (Vaadin 14 series), I think there are only two chunks - "the optimized everything that is used chunk" and the "fallback chunk". I can edit the code to make it so that the `initial` property is output to stats.json and then it is considered when adding the chunks to the bootstrap page, but I'm not sure if it really makes a difference or will everything be still to the same chunk - unless we add support to 14-series for splitting the bundle further. I doubt that it would be a priority - but I can help you get forward in trying that if you want to @xdenser

**Comment by xdenser** [comment]
OK I will try to create reduced example. I just want to load big library only when needed. In my case this is ace editor - which otherwise would add 500Kb for initial load and is only used in some rare scenarios in our application. But just for example we, of course, do not need ace editor. Some alert() in dynamically imported module will be enough.

**Comment by xdenser** [comment]
Ok here is example. 
run, Open About view and press the button.
And actually dynamic module loading works correctly unless webpack chunk name directive is used.
See frontend/dynamicModuleLoader.ts line 11
[my-app.zip](https://github.com/vaadin/flow/files/6296172/my-app.zip)
w/o directive the js file is loaded when you press button,
with directive it is loaded (but not executed) at boostrap.
